07_modules_libs/homework/handler.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strconv"
+	"sync"
+
+	"github.com/go-chi/chi/v5"
+)
+
+type AVantageCacheHandler struct {
+	data sync.Map
+}
+
+type AVantage struct {
+	MetaData        map[string]string      `json:"Meta Data"`
+	TimeSeriesDaily map[string]interface{} `json:"Time Series (Daily)"`
+}
+
+type AVantageResp struct {
+	Ticker     string `json:"ticker"`
+	PercDiff   string `json:"percentage_diff"`
+	FirstData  string `json:"first_data"`
+	SecondData string `json:"second_data"`
+}
+
+func (c *AVantageCacheHandler) Get(w http.ResponseWriter, r *http.Request) {
+
+	apiKey := "ODFNBOJQFCDO91TF"
+	ticker := chi.URLParam(r, "ticker")
+	firstDate := chi.URLParam(r, "first_date")
+	secondDate := chi.URLParam(r, "second_date")
+
+	if _, ok := c.data.Load(ticker); ok {
+		log.Println("from cache")
+		writeResponse(w, http.StatusOK, resp(c, firstDate, secondDate, ticker))
+		return
+	}
+
+	url := fmt.Sprintf("https://www.alphavantage.co/query?function=TIME_SERIES_DAILY&symbol=%s&apikey=%s", ticker, apiKey)
+
+	res, err := http.Get(url)
+
+	if err != nil {
+		log.Fatal(err)
```

Comment:
This line will terminate whole process. We should handle the error and response with non 200 status code.